test/jasmine/tests/gl_plot_interact_test.js

```diff
@@ -942,60 +946,53 @@ describe('Test gl2d plots', function() {
     it('should be able to toggle visibility', function(done) {
         var _mock = Lib.extendDeep({}, mock);
 
-        // a line object + scatter fancy
-        var OBJECT_PER_TRACE = 2;
-
-        var objects = function() {
-            return gd._fullLayout._plots.xy._scene2d.glplot.objects;
-        };
-
         Plotly.plot(gd, _mock)
         .then(delay(20))
         .then(function() {
-            expect(objects().length).toEqual(OBJECT_PER_TRACE);
-
             return Plotly.restyle(gd, 'visible', 'legendonly');
         })
         .then(function() {
-            expect(objects().length).toEqual(OBJECT_PER_TRACE);
-            expect(objects()[0].data.length).toEqual(0);
+            expect(readPixel(gd.querySelector('.gl-canvas-context'), 108, 100)[0]).toBe(0);
```

Comment:
@dfcreative Non-blocking for the first regl release, but it would be nice to make the `visible: 'legendonly'` state **NOT** clear the gl context to speed up trace visibility toggle via legend items.